subprojects/workers/src/main/java/org/gradle/workers/WorkerParameters.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Marker interface for parameter objects to {@link WorkerExecution}s.
+ *
+ * <p>
+ *     Parameter types should be interfaces, only declaring getters for {@link org.gradle.api.provider.Property}-like objects.
+ *     Example:
+ * </p>
+ * <pre class='autoTested'>
+ * public interface MyParameters extends WorkerParameters {
+ *     String getStringParameter();
+ *     void setStringParameters(String stringParameter);
+ *     ConfigurableFileCollection getFiles();
+ * }
+ * </pre>
+ *
+ * @since 5.6
+ */
+@Incubating
+public interface WorkerParameters {
+    /**
+     * Used for worker executions without parameters.
+     *
+     * <p>When {@link None} is used as parameters, calling {@link WorkerExecution#getParameters()} throws an exception.</p>
```

Comment:
Any idea why we do this? I see it's the same in artifact transforms.

Reply:
I don't know for sure without playing with it, but I suspect it's because the class generator can't correctly generate a managed object without any fields.  I don't think there's a reasonable use case where you might want to call `getParameters()` on an execution (or transform) with a `None` parameter type.

Reply:
A little investigation shows that if we relax some of the constraints on `None` such that we can generate an instance of it, the managed object doesn't get generated with a `Object[]` constructor and we blow up when trying to rehydrate the object during isolation.  There are obviously a number of options for "fixing" that, but I'd be hesitant to invest much effort without a concrete use case where doing so would be valuable.